### PR TITLE
Audit updates and fixes for admin settings and widget configurations

### PR DESCRIPTION
**Pull request by OPS-PIvers**
This pull request introduces several improvements and refactorings across the codebase, focusing on component modularity, test coverage, and UI flexibility. The most significant changes include refactoring import paths for PLC components to use absolute imports, adding and enhancing tests for widgets (especially around state and debouncing), and improving the configurability of shared UI components like `TypographySettings`. Additionally, some obsolete or unnecessary code has been removed, and minor UI fixes and label updates have been made.

**Component Refactoring and Import Path Updates:**

- Refactored multiple PLC-related components to use absolute import paths instead of relative ones, improving maintainability and reducing import errors. This affects files such as `NotesBody.tsx`, `PlcQuizLibraryBody.tsx`, `PlcVideoActivitiesBody.tsx`, `TodosBody.tsx`, `PlcDocsBody.tsx`, `PlcHome.tsx`, `PlcSettingsTab.tsx`, and others. [[1]](diffhunk://#diff-2a2a42e231f5677021ba905b0c21d27e98afe51234046b0c191922210e571a40L27-R27) [[2]](diffhunk://#diff-20ac4238dbc35ec19b7e8de8c7730f8164bbe8b75bb4e8f21220365202a7edbaL83-R85) [[3]](diffhunk://#diff-0b5da326ebc5decfa5e8f5e79c6108550470f27f7a5fa3d6ce798eec889b26fbL84-R86) [[4]](diffhunk://#diff-eb0538d74f62f2d2fe4b40b77d69d86d668a757b21003f6390164f4861f44121L10-R10) [[5]](diffhunk://#diff-315f2b51ea0449f4611f24394e7b92b0a3b117f1510a387276b063cc91061939L29-R29) [[6]](diffhunk://#diff-f010e5afb873971afca2ade7e2af15593913e738724d186d47773db21155b419L33-R34) [[7]](diffhunk://#diff-d87cb8298adb7d945737d5f3be44f7c05d5a7a376d0e3aec98b57978d1594cbeL21-R21)

**Widget and UI Improvements:**

- Enhanced the `TypographySettings` component to accept a `showColorPicker` prop, allowing widgets that manage their own color system to hide the color picker, increasing reusability. [[1]](diffhunk://#diff-d28f37536097dd9e4b5ded811b864132c6c47e67ee70964c770f0d570195c843R11-R25) [[2]](diffhunk://#diff-d28f37536097dd9e4b5ded811b864132c6c47e67ee70964c770f0d570195c843R63) [[3]](diffhunk://#diff-d28f37536097dd9e4b5ded811b864132c6c47e67ee70964c770f0d570195c843R94)
- Updated the `ClockWidget` settings to use the shared `TypographySettings` component for font family selection, and adjusted tests and code accordingly. [[1]](diffhunk://#diff-e150a1314f5c9a185ac625393198bdd4a0a5bd85bb588ca3ed895e5f3e662dc7L5-R8) [[2]](diffhunk://#diff-e150a1314f5c9a185ac625393198bdd4a0a5bd85bb588ca3ed895e5f3e662dc7L50-L60) [[3]](diffhunk://#diff-6283a5e7b4766e93c3b74642db5a4de740bc91a8d03e7663ec01085bb84a0fd4L96-R98)

**Testing Enhancements:**

- Added a comprehensive test suite for the `ActivityWallShareModal` to verify that form state resets correctly on remounts and activity changes, ensuring consistent user experience.
- Added and improved tests for the `ChecklistWidget` and its settings, specifically focusing on debounced saves, state updates, and no-op detection when input matches existing state. [[1]](diffhunk://#diff-65496d48bf33bf3f0e9564eb11282da51819d86961f6eed2e64bbfe51a7013acL2-R3) [[2]](diffhunk://#diff-65496d48bf33bf3f0e9564eb11282da51819d86961f6eed2e64bbfe51a7013acR269-R355)

**Checklist Widget Refactoring:**

- Simplified the `ChecklistSettings` component by removing an unnecessary `updateWidgetRef` indirection, as `updateWidget` is already stable. Updated the debounced save logic to use the direct reference. [[1]](diffhunk://#diff-7a698cb6cdca2c5a2b58d8b3d500afe6871897cc95302a6572b190f870aeddb3L65-R70) [[2]](diffhunk://#diff-7a698cb6cdca2c5a2b58d8b3d500afe6871897cc95302a6572b190f870aeddb3L103-R101)

**Admin and Analytics UI:**

- Improved the admin users view by enabling the "Resend invite" button for selected invited users and making it functional, instead of showing a disabled placeholder.
- Removed the obsolete 'guided-learning' label from `AI_FEATURE_LABELS` in analytics.

### CODE REVIEW

File: components/admin/Organization/views/UsersView.tsx [hunk 1]
```diff
@@ -378,15 +378,27 @@ export const UsersView: React.FC<Props> = ({
...
-                Resend invite
-              </button>
+              {(() => {
+                const invitedSelected = Array.from(selected)
+                  .map((id) => users.find((u) => u.id === id))
+                  .filter(
+                    (u): u is UserRecord =>
+                      u !== undefined && u.status === 'invited'
+                  );
+                return (
```
[the remark this answers is not in the document]
Comment (OPS-PIvers) [reply components/admin/Organization/views/UsersView.tsx:385]: Valid — applied in `6573248`. Replaced the `Array.from(selected).map(find).filter(typeguard)` (O(N·M)) with a single `users.filter((u) => selected.has(u.id) && u.status === 'invited')` pass: O(M) with an O(1) `Set` lookup, and it drops the `UserRecord` type guard. (Pushed via #2073 since direct git push to `dev-paul` is org-policy blocked.)

---
_Generated by [Claude Code](https://claude.ai/code)_

File: components/widgets/QuizWidget/components/QuizManager.tsx [hunk 2]
```diff
@@ -729,13 +735,19 @@ export const QuizManager: React.FC<QuizManagerProps> = ({
...
-  // `order` field today, so we disable drag at the grid level. The hook still
-  // mirrors items + commits nothing, preserving the primitive API.
+  const onReorderCommit = useCallback(
+    async (orderedIds: string[]) => {
+      if (onReorderQuizzes) {
+        await Promise.resolve(onReorderQuizzes(orderedIds));
+      }
+    },
+    [onReorderQuizzes]
+  );
```
[the remark this answers is not in the document]
Comment (OPS-PIvers) [reply components/widgets/QuizWidget/components/QuizManager.tsx:748]: Valid — applied in `6573248`. `onReorderCommit` now `await`s `onReorderQuizzes(orderedIds)` directly; `await` already handles both sync/void and promise returns, so the `Promise.resolve(...)` wrapper was redundant.

---
_Generated by [Claude Code](https://claude.ai/code)_

File: components/widgets/QuizWidget/components/QuizManager.tsx [hunk 3]
```diff
@@ -1320,6 +1332,16 @@ export const QuizManager: React.FC<QuizManagerProps> = ({
...
 
+  // ─── Card-to-card reorder drop handler ───────────────────────────────────
+  const handleReorderDrop = useCallback(
+    async (nextOrderedIds: string[]): Promise<void> => {
+      if (!onReorderQuizzes) return;
+      if (libraryView.reorderLocked) return;
+      await Promise.resolve(onReorderQuizzes(nextOrderedIds));
+    },
+    [libraryView.reorderLocked, onReorderQuizzes]
+  );
```
[the remark this answers is not in the document]
Comment (OPS-PIvers) [reply components/widgets/QuizWidget/components/QuizManager.tsx:1346]: Valid — applied in `6573248`. `handleReorderDrop` now `await`s `onReorderQuizzes(nextOrderedIds)` directly, dropping the redundant `Promise.resolve(...)`.

---
_Generated by [Claude Code](https://claude.ai/code)_

File: components/widgets/Checklist/Settings.tsx [hunk 4]
```diff
@@ -100,7 +98,7 @@ export const ChecklistSettings: React.FC<{ widget: WidgetData }> = ({
         });
 
       setSkipNextSync(true);
-      updateWidgetRef.current(widget.id, {
+      updateWidget(widget.id, {
```
[the remark this answers is not in the document]
Comment (OPS-PIvers) [reply components/widgets/Checklist/Settings.tsx]: Valid — fixed in `6573248`. You're right: `useDashboard().updateWidget` resolves to the `DashboardContext` value, where it's a `useCallback(..., [saveWidgetConfig])`, and `saveWidgetConfig` itself is `useCallback(..., [user])`. So its identity can change (e.g. on the hourly Google token refresh that swaps the `user` object), and the 500 ms debounced-save closure could fire with a stale callback. The genuinely-stable `stableActions.updateWidget` is only reachable via `useDashboardActions()`, which this component doesn't consume.

I restored the `updateWidgetRef` pattern (option 1) — mirrored in the render body alongside the existing `configRef`/`itemsRef`, and the timeout now calls `updateWidgetRef.current(...)`. Also corrected the misleading comment.

---
_Generated by [Claude Code](https://claude.ai/code)_

File: components/widgets/QuizWidget/Widget.tsx [hunk 5]
```diff
@@ -765,6 +768,29 @@ export const QuizWidget: React.FC<{ widget: WidgetData }> = ({ widget }) => {
...
+          order: index,
+          updatedAt: Date.now(),
+        });
+      });
+      try {
+        await batch.commit();
+      } catch (err) {
+        console.error('[QuizWidget] Failed to persist reorder:', err);
+        throw err;
+      }
```
[the remark this answers is not in the document]
Comment (OPS-PIvers) [reply components/widgets/QuizWidget/Widget.tsx:793]: Valid — fixed in `6573248`. Went with the toast-before-`throw` option. The re-throw is intentionally kept so `useSortableReorder` still reverts the visual order, but now `addToast('Could not save the new order. Please try again.', 'error')` fires first so the teacher sees why the drag snapped back. `addToast` was added to the `useCallback` deps.

I deliberately avoided `batch.set(..., { merge: true })` here: if the quiz metadata doc really was deleted concurrently, a merge-set would resurrect an orphan doc containing only `{ order, updatedAt }`, which is worse than a clean failure.

---
_Generated by [Claude Code](https://claude.ai/code)_

File: tests/components/widgets/QuizManager.reorder.test.tsx [hunk 6]
```diff
@@ -0,0 +1,244 @@
...
+    const manualComparator = (a: QuizMetadata, b: QuizMetadata): number =>
+      (a.order ?? 0) - (b.order ?? 0);
+
+    const quizzes: QuizMetadata[] = [
+      makeQuizMeta({ id: 'q3', order: 2 }),
+      makeQuizMeta({ id: 'q1', order: 0 }),
+      makeQuizMeta({ id: 'q2', order: 1 }),
+    ];
+
+    const sorted = [...quizzes].sort(manualComparator);
```
[the remark this answers is not in the document]
Comment (OPS-PIvers) [reply tests/components/widgets/QuizManager.reorder.test.tsx:151]: Valid — fixed in `6573248`. Exported `SORT_COMPARATORS` from `QuizManager.tsx` and the test now derives its `manualComparator` from `SORT_COMPARATORS.manual(a, b, 'asc')`, so a regression in the production comparator (e.g. `?? Infinity` instead of `?? 0`) is now caught here. The local lambda is gone. (The constant export needs a one-line `react-refresh/only-export-components` disable, matching the existing precedent in `GuidedLearningStudentApp.tsx`.)

---
_Generated by [Claude Code](https://claude.ai/code)_

File: components/widgets/Checklist/Widget.test.tsx [hunk 7]
```diff
@@ -266,6 +266,93 @@ describe('ChecklistWidget', () => {
...
+    // Type the exact same text that's already persisted
+    fireEvent.change(textarea, { target: { value: 'Existing' } });
+
+    act(() => {
+      vi.advanceTimersByTime(500);
+    });
+
+    expect(mockUpdateWidget).not.toHaveBeenCalled();
+  });
+});
```
[the remark this answers is not in the document]
Comment (OPS-PIvers) [reply components/widgets/Checklist/Widget.test.tsx:354]: Valid — fixed in `6573248`. The "no steps" test now clicks the Sync button (scoped to the `.bg-indigo-50` Import Routine section, matching the line-496 pattern) and asserts `mockAddToast` was called with `('Active routine has no steps to import.', 'info')`, then keeps `expect(mockUpdateWidget).not.toHaveBeenCalled()`. The guard is now actually exercised instead of the assertion being trivially true on render.

---
_Generated by [Claude Code](https://claude.ai/code)_

File: components/admin/Organization/views/UsersView.tsx [hunk 8]
```diff
@@ -378,15 +378,24 @@ export const UsersView: React.FC<Props> = ({
...
-                type="button"
-                disabled
-                aria-disabled="true"
-                title="Coming soon"
-                className="text-xs font-semibold text-white/50 cursor-not-allowed"
-              >
-                Resend invite
-              </button>
+              {(() => {
+                const invitedSelected = users.filter(
```
[the remark this answers is not in the document]
Comment (OPS-PIvers) [reply components/admin/Organization/views/UsersView.tsx:382]: Not applying an automated fix here — this is a UX judgment, not a clear bug. The two review passes disagree on the variable (the earlier Gemini suggestion used `users.filter`; this one suggests `filtered.filter`), and there is a real behavioral difference in the select-then-search case: the bulk **Deactivate** and **Delete** actions both operate on `Array.from(selected)` (every selected id), so the current `users.filter(...)` keeps **Resend invite** consistent with them. Switching to `filtered.filter(...)` would silently drop selected-but-currently-hidden invited users from the resend while Deactivate/Delete still act on them. Since `selected` is only ever populated from `filtered` rows (and `filtered` is already scope-restricted for `building_admin`), there's no scope-escape risk today either way — so this is a deliberate "all selected vs. only-visible selected" decision for the author to make rather than an automated change.

---
_Generated by [Claude Code](https://claude.ai/code)_

**Review by OPS-PIvers — COMMENTED** [review]
## Automated Code Review

**Date:** 2026-06-25

### Summary
Broad, well-scoped PR: PLC import-path cleanup (relative → `@/`), a reusable `showColorPicker` prop on `TypographySettings`, `ClockWidget`/`TimeTool` settings simplified onto the shared component, a quiz reorder feature with debounced/awaited callbacks, a working bulk "Resend invite" admin action, new `Announcement` scheduled-window fields, and substantial new test coverage. The author already addressed seven earlier review threads in `6573248` (stale-closure ref restore, NOT_FOUND toast, redundant `Promise.resolve` removals, two test-correctness gaps, the O(N·M)→O(M) resend filter). This review focuses on the items raised *after* that commit.

### Code Quality
✓ The post-`6573248` state is clean. Import-path conversions and the `showColorPicker` defaulting are correct, and the `updateWidgetRef` justification comment now matches reality (`useDashboard().updateWidget` is the `DashboardContext` `useCallback(..., [saveWidgetConfig])`, not the stable `useDashboardActions()` one).

### Widget Standards
✓ Not applicable in the new-widget sense — no new widgets registered. The `TypographySettings` refactor correctly threads `showColorPicker` through without breaking existing consumers, and `ClockWidget` font-family selection now reuses the shared primitive per the appearance standard.

### Admin Config Standards
- **`UsersView` bulk "Deactivate" ungated by `canManageUsers` (line 422)** — flagged by a prior review as a "confirmed bug," but I believe this is a **false positive**. Per-row deactivation is governed by `canEditStatus = inScope` (line 574), deliberately *separate* from `canManage` (which gates edit/role/buildings/delete). So `building_admin`s are intended to deactivate in-scope users, and `selected` is only ever populated from `filtered` rows — which are already scope-restricted for `building_admin` (lines 199–202) and re-scoped by "select all" (line 462). The ungated bulk Deactivate is therefore **consistent** with the per-row permission model; wrapping it in `{canManageUsers && …}` would *remove* a legitimate capability. Recommend confirming the intended `building_admin` permission set, but no change should be made blindly.
- **`UsersView` resend filters `users` vs `filtered` (line 382)** — a UX judgment, not a bug (replied inline). `users.filter` keeps bulk Resend consistent with bulk Deactivate/Delete, which act on all `selected`.

### Firestore & Security
- ✓ No new collections without matching `firestore.rules` blocks; no auth-bypass or client-secret patterns.
- **`AnnouncementOverlay` PATH A omits the server-side `isActive` filter** (intentional, documented to avoid a composite index). Non-blocking, but worth tracking: as archived announcements accumulate, every teacher in the org streams all inactive docs per page load. Add the composite index + filter once volume warrants.

### Regression Risk
- ✓ Does **not** touch `DashboardContext.tsx`'s config-merge / `getAdminBuildingConfig` pipeline, `WidgetRegistry.ts`, or add a new widget type to `types.ts` (so the `WidgetConfig` union / `ConfigForWidget` invariant is unaffected).
- Quiz reorder uses `writeBatch.update()` (requires the doc to exist) — the concurrent-delete `NOT_FOUND` path is now handled with an explicit error toast before the re-throw (fixed in `6573248`), with the re-throw deliberately preserved so `useSortableReorder` reverts the visual order. Reasonable.
- **Due-date parsed as UTC midnight (`QuizManager.tsx` line 2300)** — flagged as a timezone bug, but the current `new Date(val).getTime()` parse is **symmetric** with the `new Date(dueAt).toISOString().slice(0,10)` display on line 2292 and matches the **documented date-only convention** in `utils/localDate.ts` (`splitDueAtToInputs(dueAt, /*hasTime*/ false)`: date-only `dueAt` is stored/surfaced as the UTC calendar date). The suggested `val + 'T00:00:00'` local-midnight change would break that symmetry for eastern-hemisphere (UTC+) users, where local midnight rolls the UTC calendar date back a day. If the project ever wants true local-date semantics here, both the parse *and* the display must move to local components together (and ideally route through `dueInputsToEpoch`/`splitDueAtToInputs`) — not a one-line change.

### Missing Tests
- ✓ Reorder comparator now tested through the exported `SORT_COMPARATORS` (regression-catching), and the Checklist "no steps" test now actually clicks Sync and asserts the toast — both hardened in `6573248`.
- Gap: the new `Announcement` scheduled-window logic has an untested edge case — a **half-populated end window** (`scheduledEndDate` set, `scheduledEndTime` blank, or vice-versa) makes `combineDateAndTime` return `null`, so `isWithinActiveWindow` silently never auto-expires. Whether to require both fields together or default the missing time (precedent: `DEFAULT_DUE_TIME = '23:59'`) is a product/design decision — worth a follow-up with a test once decided.

### Overall
**Merge readiness:** Ready with minor notes
The substantive earlier feedback is already resolved in `6573248`. Of the newer items, the Deactivate-guard "bug" and the UTC-date "bug" are both consistent-by-design on closer inspection; the remaining notes (announcement half-window handling, `AnnouncementOverlay` index growth, `OptionInput` key-sync contract) are low-severity follow-ups rather than merge blockers.

---
_This review was generated automatically. All findings should be verified by a human before merging._

---
_Generated by [Claude Code](https://claude.ai/code)_
